### ISSUE TITLE
slice_X Error in frequency == 'batch'

### ISSUE THREAD

**Issue by mohaimenz** [maxpumperla/elephas#90]
hi @maxpumperla, it seems that in Asynchronous spark worker class, we are still trying to use slice_X which does not exist in keras anymore. 
```
line 315: from keras.engine.training import slice_X
line 322: X = slice_X(x_train, batch_ids)
line 323: y = slice_X(y_train, batch_ids)
```
Can't we simply replace the above lines with the following? 

```
X  =  np.take(x_train, batch_ids)
y = np.take(y_train, batch_ids)
```

**Comment by mohaimenz** [comment]
So for bothering you too much @maxpumperla . I ended up adding slice_X function to spark_model.py.
I am closing this issue as well.
Thanks a lot for such a nice library.

**Comment by maxpumperla** [comment]
thanks. but let's leave this open until it's properly resolved

**Comment by mohaimenz** [comment]
@maxpumperla that would be great. Thanks

**Comment by mohaimenz** [comment]
@maxpumperla awesome work max. I will update from the master branch. It seems that you have modularized your code a bit more. I did not know the slice_arrays function at keras.utils.generic_utils.
I do not have any issue right now as I have got a ubuntu machine and this api works there well. I was unable to make it work in windows though.

**Comment by maxpumperla** [comment]
even better. note that I had to break the API a little, but it's better for the user (in my opinion)

**Comment by mohaimenz** [comment]
@maxpumperla yeah! I saw that and that is why I said you have modularized the spark model. That is really great. Thanks a lot for you effort for the community.
